### PR TITLE
Redux testing 

### PR DESCRIPTION
**Pull request by AnnaEmson**
### What

Main tasks is:  
- Add integration test for redux store to test interaction between actions, reducers and store. This is added to current store configuration as a preparation for further refactoring. 
- Additionally add some improvements : change the name of Create file to CreateNewCollection ,  add selectors and correct some errors pointed out when strict mode was activated. (Look for comments inline )
Add coverage report to tests 
Please be aware this is not finished task, just an iteration of improvements and more will come later as per priorities 
Screenshot 2022-02-03 at 14 23 01
 
correct lint script as it was broken before, now shows preset rules errors as per screenshot 
Screenshot 2022-02-04 at 11 17 56
this will be corrected 

Correct some obvious linting errors and reducing the errors number to 17 errors (will work on lowering this number even further) 
Screenshot 2022-02-04 at 15 24 15

### How to review
No changes to current features in Florence, tests are passing

### Who can review

Front end devs

### CODE REVIEW

File: src/app/components/notifications/NotificationItem.jsx [hunk 1]
```diff
@@ -2,12 +2,12 @@ import React from "react";
 import PropTypes from "prop-types";
 import clsx from "clsx";
 
-const NotificationItem = ({ isVisible = false, type, message, buttons = [] }) => (
-    <li className={clsx("notifications__item", { visible: isVisible }, `notifications__item--${type}`)}>
-        {message}
-        {buttons.map((button, i) => {
+const NotificationItem = props => (
+    <li key={props.id + props.type} className={clsx("notifications__item", { visible: props.isVisible }, `notifications__item--${props.type}`)}>
```
Comment (AnnaEmson) [inline src/app/components/notifications/NotificationItem.jsx:6]: removing the error lacking the key on react component

File: src/app/components/selectable-box/double-column/DoubleSelectableBox.jsx [hunk 2]
```diff
@@ -62,7 +62,7 @@ const DoubleSelectableBox = props => {
                 >
                     {props.headings[1]}
                     <Sort active={sortConfig} name="publishDate" />
-                    {props.isUpdating && <span className="selectable-box__status pull-right loader" />}
+                    {props.isUpdating && <span data-testid="loader" className="selectable-box__status pull-right loader" />}
```
Comment (AnnaEmson) [inline src/app/components/selectable-box/double-column/DoubleSelectableBox.jsx:65]: added this for easier targeting in the test for the loading status

File: src/app/config/actions.js [hunk 3]
```diff
@@ -13,13 +13,6 @@ export function setConfig(config) {
     };
 }
 
-export function addAllCollections(collections) {
```
Comment (AnnaEmson) [inline src/app/config/actions.js:16]: removing this action as it now not used

File: src/app/config/reducer.js [hunk 4]
```diff
@@ -79,19 +78,7 @@ export default function reducer(state = initialState, action) {
                 },
             };
         }
-
-        case types.MARK_COLLECTION_FOR_DELETE_FROM_ALL_COLLECTIONS: {
```
Comment (AnnaEmson) [inline src/app/config/reducer.js:83]: This is redundant

File: src/.eslintrc.json [hunk 5]
```diff
@@ -1,5 +1,5 @@
 {
-    "extends": ["eslint:recommended", "prettier", "prettier/react"],
```
Comment (AnnaEmson) [inline src/.eslintrc.json:2]: removing as prettier/react is obsolete

File: src/app/config/thunks.js [hunk 6]
```diff
@@ -55,15 +52,17 @@ export const loadCollectionsRequest = redirect => dispatch => {
...
                     });
                     break;
                 }
                 case "FETCH_ERR": {
                     const notification = {
                         type: "warning",
-                        message: FETCH_ERR("collections"),
+                        message:
+                            "There's been a network error whilst trying to get collections. You may only be able to see previously loaded information and not be able to edit any team members.",
                         isDismissable: true,
```
Comment (NoelleDL) [inline src/app/config/thunks.js:66]: Shouldn't these remain constants?

**Review by NoelleDL — APPROVED** [review]
LGTM

**Pull request merged**